server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -279,22 +304,24 @@ private static Query refEqObject(Function eq,
                                      String fqn,
                                      ObjectType type,
                                      Map<String, Object> value,
-                                     LuceneQueryBuilder.Context context) {
+                                     Context context) {
         BooleanQuery.Builder boolBuilder = new BooleanQuery.Builder();
         int preFilters = 0;
         for (Map.Entry<String, Object> entry : value.entrySet()) {
             String key = entry.getKey();
             DataType<?> innerType = type.innerType(key);
-            if (innerType == null) {
+            if (innerType == UndefinedType.INSTANCE) {
                 // could be a nested object or not part of meta data; skip pre-filtering
                 continue;
             }
             String fqNestedColumn = fqn + '.' + key;
+            var nestedRef = context.getRef(ColumnIdent.fromPath(fqNestedColumn));
```

Comment:
Might be worth changing the `refEqObject` signature to take a `ColumnIdent` as parameter. Then you can use `column.getChild` here.